File: solutions/fsharp/hamming/1/Hamming.fs
module Hamming

let distance (strand1: string) (strand2: string): int option = 
    if ((String.length strand1) = (String.length strand2)) then
        Some(List.fold2 (fun s x y -> (if x = y then s else s + 1)) 0 (Seq.toList strand1) (Seq.toList strand2))
    else
        None

File: solutions/fortran/hamming/1/hamming.f90
module hamming
  implicit none
contains

  function compute(strand1, strand2, distance)
    character(len=*) :: strand1, strand2
    integer :: distance
    logical :: compute

    integer :: i, len1, len2

    len1 = len(trim(adjustl(strand1)))
    len2 = len(trim(adjustl(strand2)))

    if (len1 == 0 .and. len2 == 0) then
        distance = 0
        compute = .true.
        return
    endif

    if (len1 == 0 .or. len2 == 0 .or. len1 /= len2) then
        distance = 0
        compute = .false.
        return
    endif

    distance = 0
    do i = 1, len1
        if (strand1(i:i) /= strand2(i:i)) then
            distance = distance + 1
        endif
    end do

    compute = .true.
  end function compute

end module hamming
